tokio/src/io/driver/mod.rs

```diff
@@ -69,38 +63,16 @@ fn _assert_kinds() {
 
 // ===== impl Driver =====
 
-#[derive(Debug)]
-/// Guard that resets current reactor on drop.
-pub(crate) struct DefaultGuard<'a> {
```

Comment:
I removed the lifetimes from the `ThreadContext` drop guards as it was unclear to me why they were necessary. I might be missing the reason why they are needed though.